### PR TITLE
Allow hover tooltip contents

### PR DESCRIPTION
**Pull request by Mati365**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Feature (ui): Added the ability to move the mouse cursor to the tooltip without it closing.
Feature (ui): Introduced support for closing tooltips using the Escape key.

---

### Additional information

Closes https://github.com/cksource/ckeditor5-commercial/issues/6039.
Related to https://github.com/cksource/ckeditor5-commercial/pull/6098

### CODE REVIEW

File: packages/ckeditor5-ui/src/tooltipmanager.ts [hunk 1]
```diff
@@ -259,6 +260,18 @@ export default class TooltipManager extends DomEmitterMixin() {
...
 
+	/**
+	 * Handles hiding tooltips on `keydown` in DOM.
+	 *
+	 * @param evt An object containing information about the fired event.
+	 * @param domEvent The DOM event.
+	 */
+	private _onKeyDown( evt: EventInfo, { key }: KeyboardEvent ) {
+		if ( key === 'Escape' && this._currentElementWithTooltip ) {
+			this._unpinTooltip();
```
Comment (oleq) [inline packages/ckeditor5-ui/src/tooltipmanager.ts:271]: I think we need to stop propagating this event. ATM Esc will close the underlying UI despite the tooltip being displayed. In such a case, it should take two Esc key presses to close the UI.

https://github.com/ckeditor/ckeditor5/assets/1099479/fa5ccfd1-89cc-4802-80c4-4300fc97e3d8

File: packages/ckeditor5-utils/src/index.ts [hunk 2]
```diff
@@ -93,6 +93,8 @@ export { default as uid } from './uid.js';
 export { default as delay, type DelayedFunc } from './delay.js';
 export { default as verifyLicense } from './verifylicense.js';
 export { default as wait } from './wait.js';
+export { default as sanitizeHTML } from './sanitizeHTML.js';
```
Comment (oleq) [inline packages/ckeditor5-utils/src/index.ts:96]: This is not related to the PR.

File: packages/ckeditor5-utils/src/sanitizeHTML.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/**
```
Comment (oleq) [inline packages/ckeditor5-utils/src/sanitizeHTML.ts:1]: This is not related to the PR.

File: packages/ckeditor5-utils/tests/sanitizeHTML.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/**
```
Comment (oleq) [inline packages/ckeditor5-utils/tests/sanitizeHTML.js:1]: This is not related to the PR.

**Review by oleq — CHANGES_REQUESTED** [review]
Can we delay the hiding of the tooltip? It's very hard to move a cursor to it because it hides right away unless the cursor moves perfectly along the "^" (arrow pseudo-element). See the screencast below.

https://github.com/ckeditor/ckeditor5/assets/1099479/428399ce-b4db-4693-89d6-7095300a5bc7

File: packages/ckeditor5-utils/src/index.ts (hunk 2 above)
Comment (Mati365) [reply packages/ckeditor5-utils/src/index.ts:96]: Related to second PR in commercial repo, updated description.

File: packages/ckeditor5-ui/src/tooltipmanager.ts (hunk 1 above)
Comment (Mati365) [reply packages/ckeditor5-ui/src/tooltipmanager.ts:271]: Yep, `.stopPropagation` solves issue.

**Review by oleq — CHANGES_REQUESTED** [review]
The logic responsible for showing tooltips is broken after recent changes

https://github.com/ckeditor/ckeditor5/assets/1099479/6d312728-e5a6-42fd-93fd-8e6f5a8b98fb

**Comment by Mati365** [comment]
@oleq Fixed, can you check again?
